tests/test_config_parsing/summary_generator.py

```diff
@@ -0,0 +1,304 @@
+"""
+Implements a hypothesis strategy for unified summary files
+(.SMSPEC and .UNSMRY) without any optional fields.
+See https://opm-project.org/?page_id=955
+"""
+from dataclasses import astuple, dataclass
+from datetime import datetime
+from enum import Enum, unique
+from typing import Any, List, Tuple
+
+import ecl_data_io
+import hypothesis.strategies as st
+import numpy as np
+from pydantic import PositiveInt, conint
+
+from .egrid_generator import GrdeclKeyword
+
+
+@st.composite
+def summary_variables(draw):
+    """
+    Generator for summary variable mnemonic, See
+    section 11.2.1 in opm flow reference manual 2022-10.
+    """
+    first_character = draw(st.sampled_from("ABFGRWCS"))
+    if first_character == "A":
+        second_character = draw(st.sampled_from("ALN"))
+        third_character = draw(st.sampled_from("QL"))
+        fourth_character = draw(st.sampled_from("RT"))
+        return first_character + second_character + third_character + fourth_character
+
+    kind = draw(st.sampled_from([1, 2, 3, 4]))
+    if kind == 1:
+        return draw(
+            st.sampled_from(
+                ["BAPI", "BOSAT", "BPR", "FAQR", "FPR", "FWCT", "WBHP", "WWCT"]
+            )
+        )
+    elif kind == 2:
+        direction = draw(st.sampled_from("IJK"))
+        return (
+            draw(st.sampled_from(["FLOO", "VELG", "VELO", "FLOW", "VELW"])) + direction
+        )
+    elif kind == 3:
+        dimension = draw(st.sampled_from("XYZRT"))
+        direction = draw(st.sampled_from(" -"))
+        return draw(st.sampled_from(["GKR", "OKR", "WKR"])) + dimension + direction
+    else:
+        second_character = draw(st.sampled_from("OWGVLPT"))
+        third_character = draw(st.sampled_from("PIF"))
+        fourth_character = draw(st.sampled_from("RT"))
+        return first_character + second_character + third_character + fourth_character
+
+
+unit_names = st.sampled_from(
+    ["SM3/DAY", "BARSA", "SM3/SM3", "FRACTION", "DAYS", "YEARS", "SM3", "SECONDS"]
+)
+
+names = st.text(
+    min_size=8, max_size=8, alphabet=st.characters(min_codepoint=65, max_codepoint=90)
+)
+
+
+@unique
+class UnitSystem(Enum):
+    METRIC = 1
+    FIELD = 2
+    LAB = 3
+
+    def to_ecl(self):
+        return self.value
+
+
+@unique
+class Simulator(Enum):
+    ECLIPSE_100 = 100
+    ECLIPSE_300 = 300
+    ECLIPSE_300_THERMAL = 500
+    INTERSECT = 700
+    FRONTSIM = 800
+
+    def to_ecl(self):
+        return self.value
+
+
+@dataclass
+class SmspecIntehead(GrdeclKeyword):
+    unit: UnitSystem
+    simulator: Simulator
+
+
+@dataclass
+class Date:
+    day: conint(ge=1, le=31)  # type: ignore
+    month: conint(ge=1, le=12)  # type: ignore
+    year: conint(gt=1901, lt=2038)  # type: ignore
+    hour: conint(ge=0, lt=24)  # type: ignore
+    minutes: conint(ge=0, lt=60)  # type: ignore
+    micro_seconds: conint(ge=0, lt=60000000)  # type: ignore
```

Comment:
This is super unimportant, but why the plural 's' on just these? Most datetime implementations does not have that.